test/distributed/_tensor/test_dtensor.py

```diff
@@ -804,7 +804,24 @@ def test_split_tensor(self) -> None:
                     for unpadded_tensor in unpadded_list
                 ]
                 assert_array_equal(expected_is_tensor_empty, is_tensor_empty)
+    @with_comms
+    def test_metadata_consistency_check(self):
+        device_mesh = DeviceMesh(self.device_type, list(range(self.world_size)))
+        placements = [Shard(0)]
+
+        # Create a local tensor with specific metadata
+        local_tensor = torch.randn(3, 3, requires_grad=True, dtype=torch.float32)
 
+        if self.rank == 0:
+            local_tensor = local_tensor.to(dtype=torch.float64)
+
+        with self.assertRaises(ValueError):
+            DTensor.from_local(local_tensor, device_mesh, placements, run_check=True)
```

Comment:
instead of only testing `dtype` change, can we also test:
1. requires_grad change
2. shape/stride difference